tests/client.py

```diff
@@ -403,6 +425,501 @@ def get_workflows(self):
+        This is intended for generating one record for companies.
+
+        HubSpot API https://legacydocs.hubspot.com/docs/methods/companies/create_company
+        """
+        record_uuid = str(uuid.uuid4()).replace('-', '')
+        
+        url = f"{BASE_URL}/companies/v2/companies/"
+        data = {"properties": [{"name": "name", "value": f"Company Name {record_uuid}"},
+                               {"name": "description", "value": "company description"}]}
+                
+        # generate a record
+        response = self.post(url, data)
+        records = [response]
+        return records
+
+    def create_contact_lists(self):
+        """
+
+        HubSpot API https://legacydocs.hubspot.com/docs/methods/lists/create_list
+        """
+        record_uuid = str(uuid.uuid4()).replace('-', '')
+
+        url = f"{BASE_URL}/contacts/v1/lists/"
+        data = {
+            "name": f"tweeters{record_uuid}",
+            "dynamic": True,
+            "filters":[
+                [{
+                    "operator": "EQ",
+                    "value": f"@hubspot{record_uuid}",
+                    "property": "twitterhandle",
+                    "type": "string"
+                }]
+            ]
+        }
+        #TODO generate different filters 
+        # generate a record
+        response = self.post(url, data)
+        records = [response]
+        return records
+
+    def create_contacts_by_company(self):
+        """
+        TODO https://legacydocs.hubspot.com/docs/methods/companies/add_contact_to_company
+        https://legacydocs.hubspot.com/docs/methods/crm-associations/associate-objects
+        """
+        url = f"{BASE_URL}/crm-associations/v1/associations"
+        #TODO only use contacts-company combinations that do not exist yet
+        contact_records = self.get_contacts()
+        since = datetime.datetime.today()-datetime.timedelta(days=7)
+        company_records = self.get_companies(since)
+        contacts_by_company_records = self.get_contacts_by_company([company_records[0]["companyId"]])
+                
+        for company in company_records:
+            for contact in contact_records:
+                # look for a contact that is not already in the contacts_by_company list
+                if contact['vid'] not in [contacts['contact-id'] for contacts in contacts_by_company_records]:
+                    contact_id = contact['vid']
+                    company_id = company['companyId']
+
+                    data = {
+                        "fromObjectId": company_id,
+                        "toObjectId": contact_id,
+                        "category": "HUBSPOT_DEFINED",
+                        "definitionId": 2
+                    
+                    }                    
+                    # generate a record
+                    self.put(url, data)
+                    records = [{'company-id': company_id, 'contact-id': contact_id}]
+                    return records
```

Comment:
Can we get in a state where there are two `contact_id`s that are not in our list of `contact_by_company` ids?